### PR TITLE
Update schema descriptions

### PR DESCRIPTION
**Pull request by mindyls**
# Related Issues
partially fixes NRLMMD-GEOIPS/geoips#164
fixes NRLMMD-GEOIPS/geoips#233

# Testing Instructions
Run geoips/tests/integration_tests/full_test.sh to test this. Should get a return value of 0.

# Summary
Currently GeoIPS Schema Files have essentially no documentation on what these schemas
actually accomplish. There are some comments scattered throughout the code, but largely
these files have gone undocumented, leaving users without the proper knowledge of what
the plugin / family does as a whole, and the properties they encompass. This branch
adds top-level descriptions to each schema yaml file to provide the user with general
knowledge of what the schema accomplishes. A later PR will document every property
within the schema so that yaml files can be completely self-documenting.

Files Modified:

    modified: geoips/schema/bases/top.yaml
    modified: geoips/schema/product_defaults/algorithm.yaml
    modified: geoips/schema/product_defaults/algorithm_colormapper.yaml
    modified: geoips/schema/product_defaults/algorithm_interpolator_colormapper.yaml
    modified: geoips/schema/product_defaults/bases/algorithm.yaml
    modified: geoips/schema/product_defaults/bases/colormapper.yaml
    modified: geoips/schema/product_defaults/bases/coverage_checker.yaml
    modified: geoips/schema/product_defaults/bases/interpolator.yaml
    modified: geoips/schema/product_defaults/bases/unvalidated.yaml
    modified: geoips/schema/product_defaults/bases/windbarb_plotter.yaml
    modified: geoips/schema/product_defaults/interpolator.yaml
    modified: geoips/schema/product_defaults/interpolator_algorithm.yaml
    modified: geoips/schema/product_defaults/interpolator_algorithm_colormapper.yaml
    modified: geoips/schema/product_defaults/sectored_xarray_dict_to_output_format.yaml
    modified: geoips/schema/product_defaults/specs/algorithm.yaml
    modified: geoips/schema/product_defaults/specs/algorithm_colormapper.yaml
    modified: geoips/schema/product_defaults/specs/algorithm_interpolator_colormapper.yaml
    modified: geoips/schema/product_defaults/specs/interpolator.yaml
    modified: geoips/schema/product_defaults/specs/interpolator_algorithm.yaml
    modified: geoips/schema/product_defaults/specs/interpolator_algorithm_colormapper.yaml
    modified: geoips/schema/product_defaults/specs/xarray_dict_to_output_format.yaml
    modified: geoips/schema/product_defaults/unmodified.yaml
    modified: geoips/schema/product_defaults/unsectored_xarray_dict_area_to_output_format.yaml
    modified: geoips/schema/product_defaults/unsectored_xarray_dict_to_output_format.yaml
    modified: geoips/schema/product_defaults/xarray_dict_to_output_format.yaml
    modified: geoips/schema/products/algorithm.yaml
    modified: geoips/schema/products/algorithm_colormapper.yaml
    modified: geoips/schema/products/algorithm_interpolator_colormapper.yaml
    modified: geoips/schema/products/bases/product.yaml
    modified: geoips/schema/products/interpolator.yaml
    modified: geoips/schema/products/interpolator_algorithm.yaml
    modified: geoips/schema/products/interpolator_algorithm_colormapper.yaml
    modified: geoips/schema/products/list.yaml
    modified: geoips/schema/products/sectored_xarray_dict_area_to_output_format.yaml
    modified: geoips/schema/products/sectored_xarray_dict_to_output_format.yaml
    modified: geoips/schema/products/single.yaml
    modified: geoips/schema/products/unmodified.yaml
    modified: geoips/schema/products/unsectored_xarray_dict_area_to_output_format.yaml
    modified: geoips/schema/products/unsectored_xarray_dict_to_output_format.yaml
    modified: geoips/schema/products/xarray_dict_to_output_format.yaml
    modified: geoips/schema/sectors/generated.yaml
    modified: geoips/schema/sectors/metadata_families/atmosriver.yaml
    modified: geoips/schema/sectors/metadata_families/pyrocb.yaml
    modified: geoips/schema/sectors/metadata_families/static.yaml
    modified: geoips/schema/sectors/metadata_families/stitched.yaml
    modified: geoips/schema/sectors/metadata_families/tc.yaml
    modified: geoips/schema/sectors/metadata_families/volcano.yaml
    modified: geoips/schema/sectors/specs/area_definition.yaml
    modified: geoips/schema/sectors/specs/center.yaml
    modified: geoips/schema/sectors/static.yaml

### CODE REVIEW

**Review by evrose54 — DISMISSED** [review]
These updates provide a clear description of what each schema YAML entails. Good to go in my opinion.

**Review by cpcamacho — APPROVED** [review]

File: geoips/schema/product_defaults/bases/colormapper.yaml [hunk 1]
```diff
@@ -16,7 +20,7 @@ properties:
       arguments:
         type: object
         unevaluatedProperties: false
-        properties:  # These are all optional arguments, matches "matplotlib" family
+        properties: # These are all optional arguments, matches "matplotlib" family
```
Comment (mindyls) [inline geoips/schema/product_defaults/bases/colormapper.yaml:23]: Need to add descriptions for every field, so schema are self documenting.

Comment (biosafetylvl5) [reply geoips/schema/product_defaults/bases/colormapper.yaml:23]: Do you mean that we need more comments like this?

Comment (jsolbrig) [reply geoips/schema/product_defaults/bases/colormapper.yaml:23]: We can do this in another PR or series of PRs, but the idea would be to add `description:` to every property.  So, `data_range:` might look like this:
```
          data_range:
            description: "The minimum and maximum data values to use in normalization or clipping."
            type: array
            minItems: 2
            maxItems: 2
            items:
              type: number

**Comment by jsolbrig** [comment]
Sorry that this PR has been languishing for so long. I think I have an idea of what needs to be done with this. It is going to require some conversation between @evrose54 @biosafetylvl5 and myself to hash out the details but, in brief, I would like to do the following:

- Collect the descriptions used for the various top-level plugin schema and move them onto the docstrings for either their respective Interfaces or their Plugin classes.
- Update the current description to describe these documents as:
  - a schema
  - that validates the structure of _________
- Create a new issue to add descriptions to each property in each schema.

Before proceeding on any of this, we should discuss exactly how we want this done and whether it achieves our goals.

**Comment by biosafetylvl5** [comment]
Has this PR been made irrelevant by us switching to pydantic? Since pydantic has a `description` field in its schema definitions?

**Comment by biosafetylvl5** [comment]
Yes - closing this PR for now; some of the content might be helpful in the future but this PR is moot

**Pull request closed without merging**